### PR TITLE
Fix apigee_edge_teams_cron() if run without teams

### LINKED ISSUE #648: PHP error on `/user/%user/team-invitations` when there is no team in the organization
## Description
When there is no team in the Apigee Edge organization, the `/user/%user/team-invitations` page throws an error: "The website encountered an unexpected error. Please try again later."

## Steps to Reproduce
1. Remove all the teams from the organization (eg. as uid=1 from the `/teams` page).
2. Clear the caches.
3. Visit the `/user/%user/team-invitations` page as an authenticated user (even as uid=1).

## Actual Behavior
An error is displayed, and there is one single line in the logs:
```
Drupal\Core\Database\InvalidQueryException: Query condition 'team_invitation.team IN ()' cannot be empty. in Drupal\Core\Database\Query\Condition->condition() (line 105 of /mnt/files/local_mount/build/web/core/lib/Drupal/Core/Database/Query/Condition.php).
```

## Expected Behavior
No error is displayed and logged; instead, either a "There are no teams yet." or a "You do not have any invitations." message should be displayed.

## Screenshots
(none)

## Notes
It doesn't matter if the organization had any teams earlier or not. The bug only appears when caches are cleared AFTER all the teams have been deleted.

There's a mitigation: if at least one team exists in the organization, the problem disappears, even if there is no invitation to or membership in this team at all.

## Version Info
```
apigee/apigee-client-php                                  2.0.6         Client library for connecting to the Apigee Edge API.
drupal/apigee_edge                                        1.22.0        Apigee Edge for Drupal.
```

### LINKED ISSUE #660: Apigee Edge Teams has problems with no teams
## Description
```
$ drush cron
 [notice] Starting execution of apigee_edge_cron().
 [notice] Starting execution of apigee_edge_teams_cron(), execution of apigee_edge_cron() took 1.34ms.
 [error]  Drupal\Core\Database\InvalidQueryException: Query condition 'team_invitation_2.team IN ()' cannot be empty. in Drupal\Core\Database\Query\Condition->condition() (line 105 of /path/to/project/web/core/lib/Drupal/Core/Database/Query/Condition.php).
 [notice] Starting execution of comment_cron(), execution of apigee_edge_teams_cron() took 155.02ms.
 [notice] Starting execution of dblog_cron(), execution of comment_cron() took 2.99ms.
 [notice] Starting execution of field_cron(), execution of dblog_cron() took 0.7ms.
 [notice] Starting execution of file_cron(), execution of field_cron() took 0.88ms.
 [notice] Starting execution of history_cron(), execution of file_cron() took 2.89ms.
 [notice] Starting execution of node_cron(), execution of history_cron() took 0.55ms.
 [notice] Starting execution of search_cron(), execution of node_cron() took 7.38ms.
 [notice] Starting execution of system_cron(), execution of search_cron() took 6.01ms.
 [notice] Starting execution of update_cron(), execution of system_cron() took 13.69ms.
 [notice] Execution of update_cron() took 8.86ms.
 [notice] Cron run completed.
$
```

## Steps to Reproduce
1. Install Apigee Edge Teams module on a site connected to Apigee Edge which does not have any team added just yet.
2. Run cron (eg. via `drush cron` as above).

The other way to trigger the problem is to visit the `user/%user/team-invitations` page (without having any teams in Apigee Edge just yet).

## Actual Behavior
Invalid SQL was generated as the `IN` condition cannot accept empty sets.

## Expected Behavior
Cron should have finished without errors.

## Screenshots
See above.

## Notes
I could manage to track it down to `\Drupal\apigee_edge_teams\EventSubscriber\TeamInvitationQueryAccessSubscriber::onQueryAccess()`: as no `$team_ids` were found, the `->addCondition()` call is made with an empty array, which translates to `IN ()` triggering the error. If no teams are found (eg. because there is none in Apigee Edge), then an always false condition should be added to the query.

## Version Info
```
drupal/apigee_edge                                        1.25.0        Apigee Edge for Drupal.
```
Based on my findings above, this bug is also present on the 2.x branch, and it has nothing to do with the version of Drupal core.

### PR DESCRIPTION
**Pull request by boobaa**
closes #660

### CODE REVIEW

File: modules/apigee_edge_teams/src/EventSubscriber/TeamInvitationQueryAccessSubscriber.php [hunk 1]
```diff
@@ -66,7 +66,14 @@ public function onQueryAccess(QueryAccessEvent $event) {
...
     // the team might have been deleted on the remote server.
     $team_ids = array_keys($this->entityTypeManager->getStorage('team')->loadMultiple());
-    $event->getConditions()->addCondition('team', $team_ids);
+    if ($team_ids) {
+      $event->getConditions()->addCondition('team', $team_ids);
+    }
+    else {
+      // When no teams could be loaded from the storage, no invitation should be
+      // returned at all.
+      $event->getConditions()->alwaysFalse();
```
Comment (mxr576) [inline modules/apigee_edge_teams/src/EventSubscriber/TeamInvitationQueryAccessSubscriber.php:75]: Looks good to me, I do not see a direct test coverage for this class in the code, which concerns me. Maybe it is time to add one?

**Review by mxr576 — APPROVED** [review]

**Review by phdhiren — APPROVED** [review]
LGTM

File: modules/apigee_edge_teams/src/EventSubscriber/TeamInvitationQueryAccessSubscriber.php (hunk 1 above)
Comment (giteshk) [reply modules/apigee_edge_teams/src/EventSubscriber/TeamInvitationQueryAccessSubscriber.php:75]: Added a ticket to add test cases: #665

**Comment by giteshk** [comment]
LGTM

**Pull request merged**